### ISSUE TITLE
[DELIVERED] D03|PR864|RADA-LAWS-NORMALIZATION|REAL-EXECUTION-CONVERGENCE|V2

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1182]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: CONTINUOUS-20260911T0401Z-PR864
INSTANCE_TAG: SOL-RADA864-5J8N
SWARM_LANE_KEY: D03|PR864|RADA-LAWS-NORMALIZATION|CONVERGENCE-REAL-EXECUTION-V2
PREFERRED_WORK_KIND: REAL_EXECUTION / CI_REPAIR / CONVERGENCE
PARENT_ISSUES: #4 #548 #618 #641 #830 #864 #1049
STATUS: ACTIVE_STALE_TAKEOVER_PENDING_POSTCLAIM_ARBITRATION
PREDECESSOR_OWNER: #1049
PREDECESSOR_LEASE_EXPIRED_UTC: 2026-09-11T03:57:00Z
CLAIMED_AT_UTC: 2026-09-11T04:01:00Z
LEASE_UNTIL_UTC: 2026-09-11T10:01:00Z
INCUMBENT_PR: #864
INCUMBENT_BRANCH: d03/rada-bulk-normalization-current-main-20260907
INCUMBENT_HEAD_AT_CLAIM: 4d28dbf9002a79494badc6965623f59e4b5cb563
BASE_MAIN_AT_CLAIM: 7c14db5f6e43d369bf859cb5c6803024c929943c

STALE-TAKEOVER EVIDENCE:
- #1049 declared lease ended 2026-09-11T03:57:00Z and has no comments/renewal; issue last updated 2026-09-10T21:55:11Z.
- direct OPEN search for PR864 immediately before claim returned only expired #1049.
- incumbent #864 remains open; do not create a successor parser/framework.
- current head advanced under predecessor to 4d28dbf9002a79494badc6965623f59e4b5cb563, commit message `D03: rebind Rada normalization to live exact observation`.
- exact-head CI and specialist workflow both concluded `action_required` with zero jobs, so this is not a scientific/source failure and is not PASS.

OBJECTIVE:
Finish the incumbent Rada laws normalization lane to terminal zero-credit real execution evidence: recover exact current branch state, preserve exact #830 source/inventory authority and normalization semantics, obtain real two-clean LOCAL_FREE normalization over the pinned official archive, retain compact text-free identities, remove any temporary execution workflow, and obtain terminal shared exact-head CI plus independent audit handoff.

OWNED SURFACES:
- PR #864 existing branch only;
- exact CI/execution unblock without semantic widening;
- exact pinned archive/member inventory verification and deterministic two-clean normalization;
- compact hash/count/environment evidence only;
- final cleanup of temporary specialist workflow if present.

AVOID SURFACES:
- no Rada_Trees #925/#1017 mutation; no #1055 global-dedup Product mutation; no #1172 integration ownership;
- no privacy/decontamination/balance/split/packing/unique-loss algorithm changes;
- no tokenizer fit, optimizer/model training, final-test access, paid compute, external LLM/API data construction, or foreign pretrained weights.

ACCEPTANCE:
1. postclaim direct OPEN arbitration confirms no earlier valid current PR864 successor after #1049 expiry; otherwise yield before Product mutation;
2. preserve exact official archive SHA-256 9f43da3ffe0855e49ca47ddf906da9e7c40644a845b345f6c4a19b39ef5eaf22, 46,700,063 bytes, 3,052 canonical d*.htm objects / 353,354,780 raw HTML bytes, inventory identity 766e54cf706453de30136e02f0361c549c2b96aba9855ee0573d28b354ad077b unless exact live authority proves predecessor claim stale;
3. two independent clean normalizations are byte-identical or publish exact non-fabricated negative evidence;
4. durable evidence contains no source text and remains NORMALIZED_CANDIDATE_ZERO_CREDIT;
5. no temporary specialist workflow survives final Product head;
6. shared CI on final exact head is terminal SUCCESS; fresh independent audit required before integration.

TRUTH_BOUNDARY:
REAL_DATA_EXECUTED_BY_THIS_TAKEOVER=false
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
FINAL_TEST_ACCESSED=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false

**Comment by Oleksii-debug** [comment]
POSTCLAIM ARBITRATION COMPLETE: direct OPEN `PR864` collection immediately after #1182 creation returned only expired predecessor #1049; no newer valid current successor was present. Takeover is ACTIVE on incumbent PR #864 only. Late-binding correction: predecessor already rebound the live mutable Rada archive to exact current observation `archive_sha256=0b9e8ed8fe8aa663a68d2bc4eba858a754c626391dd7b5c461d50c3b6260df63`, `archive_bytes=46,709,153`, `canonical_entry_count=3,052`, `canonical_raw_bytes=353,426,370`, `entry_identity_sha256=3939c8a407a910222173bba5a2df1b9c7d2edce1503c13b31a9c293d61c3a603`; stale #830 values are not forced. Current exact head `4d28dbf9002a79494badc6965623f59e4b5cb563` contains that rebind plus a temporary workflow. Its CI/specialist runs are `action_required` with zero jobs after a bot-pushed head; neither is scientific failure nor PASS. Next mutation preserves the current exact observation and repairs only the execution/CI surface.

**Comment by Oleksii-debug** [comment]
PRODUCT LANE DELIVERED — exact-head handoff

Final PR: #864
Final Product head: `50069882dfb5866946626720a997a2c35458df7c`
Live main rechecked: `7c14db5f6e43d369bf859cb5c6803024c929943c`
Shared CI: run `34561463792` = terminal SUCCESS (workflow budget + Ruff + full pytest all PASS).

REAL LOCAL_FREE execution witness:
- execution head `b2088754aa2d5ed6059d89587bbbf08437ba0f55`
- run `34561144712` = SUCCESS
- artifact `10184406901`, digest `sha256:13e344a5a630f6260fdfa09684dbe73abe3ab0d30b724809938d51b03bfa2f1e`
- durable text-free report `docs/reports/d03_rada_bulk_normalization_real_execution_v1.json`
- durable evidence identity `663a8d60b595a6a57fc5fb5be631ce4ebb791e00ffc2bb6de8eae8d2c9743d30`

Verified two-clean identities:
- official archive 46,709,153 bytes, sha256 `0b9e8ed8fe8aa663a68d2bc4eba858a754c626391dd7b5c461d50c3b6260df63`
- 3,052 canonical objects / 353,426,370 raw bytes
- pinned report sha256 `9d94674323414d30f18517a71edbfee839a27a09a076eb94fe94d74cbaab53c8`
- 3,052/3,052 nonempty normalized records
- 211,176,449 normalized bytes observed, **0 credited**
- normalized JSONL sha256 `b46baa0f1c5087f4a9772ee273da459e87c0e82111dd5e51b22fc0b85cde840e`
- manifest identity `ee1c59dbc481b83bffb1380f751880e5fa6e64d654b5abad266c3e2c4d18293b`
- both source downloads, pins, JSONLs, manifests byte-identical.

Temporary execution workflow is absent from final head. PR body has been rebound to the live observation/evidence.

Truth boundary remains `NORMALIZED_CANDIDATE_ZERO_CREDIT_TWO_CLEAN_REAL_EXECUTION`: current_corpus_eligible=false; training_authorized=false; training_authorized_bytes=0; authorized optimized-target exposure=0; tokenizer fit=false; training=false; learned weights=false; final test=false; paid compute=false; foreign pretrained weights=false.

Fresh independent audit is already claimed on exact head (#1185; later duplicate #1187 was collision-noted). Product mutation ownership is released unless a canonical auditor returns CHANGES_REQUIRED.

**Comment by Oleksii-debug** [comment]
Independent audit #1185 completed on PR #864 exact head `50069882dfb5866946626720a997a2c35458df7c`: **CHANGES_REQUIRED_AUTHORITY_BINDING** (`AUDIT864-001 HIGH`).

The real run is genuine: I freshly replayed observation artifact `10175280333` and execution artifact `10184406901`; all claimed source/run/two-clean identities match, including 3,052 records / 211,176,449 normalized observed-not-credited bytes. Execution head -> final head changes only workflow removal + durable evidence add.

Blocker: final `tools/normalize_d03_rada_bulk_html.py` never consumes/validates `successor_observation_pin`; `probe_report_sha256` is only format-checked and embedded, not compared to frozen `9d946743...`; `_validate_probe()` accepts any internally consistent pinned-looking report/archive matching generic parent worker/config/family fields. A self-consistent substitute archive/report can therefore reach `PASS_NORMALIZATION_ONLY` under the production config. The temporary real-run workflow itself hard-asserted `9d946743...`, so the historical witness remains truthful; reusable integration is fail-open.

Same-lineage repair requested on #864: hard-bind report SHA + archive SHA + inventory identity to the frozen successor pin, validate successor pin closed-world/strict types, add forged substitute report+archive regression plus bool-as-zero regression, exact-head CI, then fresh bounded two-clean LOCAL_FREE witness on repaired code and fresh audit. Canonical detailed finding: #864 comment `5629408848`.

Do not grant corpus/training credit; optimized exposure remains 0, no training/final-test/paid compute/foreign weights.

**Comment by Oleksii-debug** [comment]
CODE-LEVEL REPAIR DETAIL for canonical `AUDIT864-001` (no Product mutation by me; #1182 owns #864).

Fresh exact read of final head `50069882...` confirms the narrow fail-open path and a minimal fail-closed patch shape:

`tools/normalize_d03_rada_bulk_html.py::_validate_config()` currently validates `parent_probe`, normalization policy and claim boundary, but never validates/consumes `config["successor_observation_pin"]`. `materialize_normalized_records()` only format-checks the caller-supplied `probe_report_sha256`; `_validate_probe()` verifies internal probe/archive consistency against the generic old parent contract. Hence a self-consistent substitute can pass.

Recommended same-lineage runtime binding before `_validate_probe()`/materialization:
1. require `successor_observation_pin` Mapping with a **closed-world exact key set** matching the production contract;
2. exact-bind immutable scalar authority from current config: `source_pr=864` as exact int/not bool, `source_head_sha=538fb05699fd9f61b31c0e3552faf9b144a6d3bb`, exact `config_path`, `config_identity_sha256=aed83e...`, `source_observation_report_sha256=e2ad0d...`, `pinned_probe_report_sha256=9d946743...`, `pin_status=PASS_EXACT_EXECUTION`, `archive_sha256=0b9e8ed8...`, `entry_identity_sha256=3939c8a4...`, `discovery_capacity_threshold=FAIL_BELOW_MINIMUM`, `normalization_input_only is True`;
3. require exact-int/not-bool zeros for successor-pin `normalized_capacity_credited` and `training_authorized_bytes`;
4. require caller `probe_report_sha256 == successor_pin["pinned_probe_report_sha256"]` (not just SHA syntax);
5. require `_sha256(archive) == successor_pin["archive_sha256"]`;
6. after `_validate_probe()` recomputes inventory identity, require probe inventory `entry_identity_sha256 == successor_pin["entry_identity_sha256"]` (and preferably exact count/raw-total if frozen in the pin config authority);
7. fix the same Python bool/int alias in `_validate_config()` claim-boundary `training_authorized_bytes`: current `!= 0` accepts `False`; use `type(value) is int and value == 0`. `_validate_probe()` has the same `probe.get("training_authorized_bytes") != 0` alias and should be strict too.

Adversarial regressions should construct a fully self-consistent alternate probe + alternate ZIP and prove rejection specifically on frozen pin mismatch; separately mutate caller report hash, archive, inventory identity, pin status/booleans, and `training_authorized_bytes=False` while resealing whatever self-hashes are under test.

This patch should preserve the genuine historical run as history only. Because scientific runtime bytes change, canonical #1185 correctly requires a fresh two-clean LOCAL_FREE witness on repaired code, then workflow removal/final exact-head shared CI and fresh auditor.
